src/netcdf.nim

```diff
@@ -1,2 +1,20 @@
 import netcdf/netcdf_bindings
 export netcdf_bindings
+
+type
+  NcId* = distinct cint
+
+  NetcdfError* = object of CatchableError
+
+template handleError(body: untyped) =
+  let retval: cint = body
+  if retval != 0:
+    raise newException(NetcdfError, "Error " & $retval & ": " & $strerror(retval))
+
+proc `$`*(ncid: NcId): string = $ncid.int
+
+proc open*(path: string, mode = NC_NOWRITE): NCid =
```

Comment:
I'm not familiar with the possible `mode` options `netCDF` provides. In HDF5 multiple "mode" like arguments may be combined. In that case I define an optional (more on that below) `set` of possible "mode" like flags.
That `set` contains elements of an `enum` that maps enum fields to the correct HDF5 constants. The constants there can be combined via bitwise OR operations, which one can conveniently do by OR'ing all elements of the set.

It is *optional*, because for more convenience I provide a default string like `"r"`, `"rw"` argument that is probably more expected for the average user coming from in case of HDF5 something like `h5py`. 

Not sure how this applies in `netCDF` though.